src/main/java/com/wavefront/dropwizard/metrics/TaggedMetricName.java

```diff
@@ -0,0 +1,247 @@
+    return registry.counter(taggedMetricName.encode());
+  }
+
+  /**
+   * Return the {@link Histogram} registered under the provided {@link TaggedMetricName}; or create
+   * and register a new {@link Histogram} if none is registered.
+   *
+   * @param registry          The metric registry to register to.
+   * @param taggedMetricName  The taggable metric name.
+   * @return a new or pre-existing {@link Histogram}.
+   */
+  public static Histogram histogram(MetricRegistry registry, TaggedMetricName taggedMetricName) {
+    return registry.histogram(taggedMetricName.encode());
+  }
+
+  /**
+   * Return the {@link Histogram} registered under the provided {@link TaggedMetricName}; or create
+   * and register a new {@link Histogram} using the provided MetricSupplier if none is registered.
+   *
+   * @param registry          The metric registry to register to.
+   * @param taggedMetricName  The taggable metric name.
+   * @param supplier          The MetricSupplier that is used to manufacture the {@link Histogram}.
+   * @return a new or pre-existing {@link Histogram}.
+   */
+  public static Histogram histogram(MetricRegistry registry, TaggedMetricName taggedMetricName,
+                                    MetricRegistry.MetricSupplier<Histogram> supplier) {
+    return registry.histogram(taggedMetricName.encode(), supplier);
+  }
+
+  /**
+   * Return the {@link Meter} registered under the provided {@link TaggedMetricName}; or create
+   * and register a new {@link Meter} if none is registered.
+   *
+   * @param registry          The metric registry to register to.
+   * @param taggedMetricName  The taggable metric name.
+   * @return a new or pre-existing {@link Meter}.
+   */
+  public static Meter meter(MetricRegistry registry, TaggedMetricName taggedMetricName) {
+    return registry.meter(taggedMetricName.encode());
+  }
+
+  /**
+   * Return the {@link Timer} registered under the provided {@link TaggedMetricName}; or create
+   * and register a new {@link Timer} if none is registered.
+   *
+   * @param registry          The metric registry to register to.
+   * @param taggedMetricName  The taggable metric name.
+   * @return a new or pre-existing {@link Timer}.
+   */
+  public static Timer timer(MetricRegistry registry, TaggedMetricName taggedMetricName) {
+    return registry.timer(taggedMetricName.encode());
+  }
+
+  /**
+   * Removes the metric with the given name.
+   *
+   * @param registry          The metric registry to remove from.
+   * @param taggedMetricName  The taggable metric name.
+   * @return whether or not the metric was removed.
+   */
+  public static boolean remove(MetricRegistry registry, TaggedMetricName taggedMetricName) {
+    return registry.remove(taggedMetricName.encode());
+  }
+
+  private static Pair<String, String>[] makeTags(Map<String, String> tags) {
+    if (tags == null) {
+      throw new IllegalArgumentException("Tags needs to be specified");
+    }
+    @SuppressWarnings("unchecked")
```

Comment:
why is this needed?

Reply:
It suppresses an Unchecked Assignment warning on the Pair[] in the subsequent line. The code that I templated off of did this -- I guess it's a little more performant than building a List and then converting to array